x/gov/keeper/msg_server.go

```diff
@@ -41,7 +42,20 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1.MsgSubmitPropos
 	if err != nil {
 		return nil, err
 	}
-	proposal, err := k.Keeper.SubmitProposal(ctx, proposalMsgs, msg.Metadata)
+
+	proposer, err := sdk.AccAddressFromBech32(msg.GetProposer())
+	if err != nil {
+		return nil, err
+	}
+
+	if msg.Title == "" {
+		return nil, goerrors.New("proposal title cannot be empty")
+	}
+	if msg.Summary == "" {
+		return nil, goerrors.New("proposal summary cannot be empty")
+	}
```

Comment:
As I can see, these checks are already present in `ValidateBasic`, do we still need here?